spec/forms/state_file/id_grocery_credit_form_spec.rb

```diff
@@ -0,0 +1,166 @@
+      let!(:intake) { create :state_file_id_intake, :single_filer_with_json }
+      let(:invalid_params) do
+        {
+          household_has_grocery_credit_ineligible_months: "yes",
+          primary_has_grocery_credit_ineligible_months: "yes",
+          primary_months_ineligible_for_grocery_credit: "",
+        }
+      end
+
+      it "is invalid and has errors for primary months ineligible" do
+        form = described_class.new(intake, invalid_params)
+        expect(form).not_to be_valid
+        expect(form.errors).to include :primary_months_ineligible_for_grocery_credit
+      end
+    end
+
+    context "without selecting a spouse ineligible months number after specifying that spouse had ineligible months" do
+      let!(:intake) { create :state_file_id_intake, :mfj_filer_with_json }
+      let(:invalid_params) do
+        {
+          household_has_grocery_credit_ineligible_months: "yes",
+          primary_has_grocery_credit_ineligible_months: "no",
+          primary_months_ineligible_for_grocery_credit: "",
+          spouse_has_grocery_credit_ineligible_months: "yes",
+          spouse_months_ineligible_for_grocery_credit: "",
+        }
+      end
+
+      it "is invalid and has errors for spouse months ineligible" do
+        form = described_class.new(intake, invalid_params)
+        expect(form).not_to be_valid
+        expect(form.errors).to include :spouse_months_ineligible_for_grocery_credit
+      end
+    end
+
+    context "without selecting a dependent ineligible months number after specifying that dependent had ineligible months" do
+      let!(:intake) { create :state_file_id_intake, :with_dependents }
+      let(:first_dependent) { intake.dependents[0] }
+      let(:second_dependent) { intake.dependents[1] }
+      let(:third_dependent) { intake.dependents[2] }
+      let(:invalid_params) do
+        {
+          household_has_grocery_credit_ineligible_months: "yes",
+          primary_has_grocery_credit_ineligible_months: "no",
+          primary_months_ineligible_for_grocery_credit: "",
+          dependents_attributes: {
+            '0': {
+              id: first_dependent.id,
+              id_has_grocery_credit_ineligible_months: "yes"
+            },
+            '1': {
+              id: second_dependent.id,
+              id_has_grocery_credit_ineligible_months: "no"
+            },
+            '2': {
+              id: third_dependent.id,
+              id_has_grocery_credit_ineligible_months: "no"
+            },
+            '3': {
+              id: first_dependent.id,
+              id_months_ineligible_for_grocery_credit: ""
+            },
+            '4': {
+              id: second_dependent.id,
+              id_months_ineligible_for_grocery_credit: ""
+            },
+            '5': {
+              id: third_dependent.id,
+              id_months_ineligible_for_grocery_credit: ""
+            }
```

Comment:
ooh super fascinating to see this is how the params come through for the dependents